nselib/httpcookies.lua

```diff
@@ -0,0 +1,476 @@
+      --[[if httponly ~= nil and httponly == true and shortport.http(host,port) == false then
+        stdnse.debug1("%s cookie doesnt match the httponly attribute", r_cookie.name)
+        flag = false
+      end]]--
+      if (flag == true) then
+        cookie_table[#cookie_table+1] = self.cookies[r_index]
+      end
+    end
+    return cookie_table
+  end,
+
+  --- This function merges the cookies received in <code>response.cookies</code>
+  -- to the cookies that already exist in the options.
+  -- The merge is based on RFC 6265 and when a different cookie with same <code>
+  -- name</code>, <code>path</code> and <code>domain</code> is received, it replaces
+  -- the old cookie, else it gets appended at the end of <code>options.cookies</code table.
+  -- @param cookies The cookies table to be appended in <code>self.cookies</code>
+  -- @return cookies The complete cookie table having new cookies appended
+  merge_cookie_table = function(self, cookies)
+    local flag = false
+    for r_index,r_cookie in pairs(cookies) do
+      flag = false
+      for o_index,o_cookie in pairs(self.cookies) do
+        if(r_cookie.name == o_cookie.name) then
+        --We need to check if domain and path are equal.
+        --Note:If both domain and path are nil for r_cookie and o_cookie,
+        --we need to change the cookie value 
+        --See RFC 6265 Section 5.3 for how duplicate cookies are handled
+          if(r_cookie.domain == o_cookie.domain and r_cookie.path == o_cookie.path and self.options.no_cookie_overwrite == false) then
+            self.cookies[o_index].value = cookies[r_index].value
+            flag = true
+            break
+          end
+        end
+      end
+      if (flag == false) then
+        self.cookies[#self.cookies+1] = cookies[r_index]
+      end
+    end
+    cookies = self.cookies
+    return cookies
+  end,
+
+  -- Sets the no_cookie_overwrite used by the httpcookies library
+  -- @param no_cookie_overwrite A boolean value for setting the option in library/
+  set_no_cookie_overwrite = function(self, no_cookie_overwrite)
+    self.options.no_cookie_overwrite = no_cookie_overwrite or false
+  end,
+
+  ---This function calls the http.get. It then parses the 
+  --cookies and merges them with the previously stored cookies.
+  --Several options can alter the behavior of the cookies library.
+  --@param host Host table
+  --@param port Port table
+  --@param path Path
+  --@param options Options table containing various options.
+  --@return Response Table with the previous cookies appended as well
+  get = function(self, host, port, path, options)
+    local response
+    --Here, the cookies present in the object will automatically be taken 
+    if options == nil then options = {} end
+    options.cookies = self.check_cookie_attributes(self, host, port, path)
+    response = http.get(host, port, path, options)
+    if response and response.status == 200 and response.cookies then 
```

Comment:
This only sets the cookies when a 200 response is received yet other status codes can set cookies too.